core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -456,6 +457,65 @@ class ZkUtils(val zkClient: ZkClient,
   }
 
   /**
+    * Callback that processes result of asynchronous version of conditionalUpdatePersistentPathAsync
+    */
+  private class conditionalUpdatePersistentPathCallback extends StatCallback {
```

Comment:
Can you make this an object rather than a class? There is no state to keep so you don't need to instantiate.